### PR TITLE
Fixes for issues found by LGTM

### PR DESCRIPTION
**Pull request by keszybz**
/cc @evverx

### CODE REVIEW

**Comment by poettering** [comment]
looks good to me, but the CI is a bit bitchy about this

**Comment by evverx** [comment]
This pull request **fixes 9 alerts** when merging 50abf4b170da977dcf6e26300466794335051c07 into 0bc7a22d9393ee3a806c308e8e1dd387858d44ba - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-ec3d67479b001c36ada7493a2e893ac90f477f9f)

**fixed alerts:**

* 4 for Potentially overflowing call to snprintf
* 3 for Comparison result is always the same
* 2 for Multiplication result converted to larger type

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by evverx** [comment]
This pull request **fixes 9 alerts** when merging 50abf4b170da977dcf6e26300466794335051c07 into 14f37112c86b573cf7f1f557e6d1be4c322290a4 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-f0d2593b428c58cd5efcd74b66c33f9b1cbe293f)

**fixed alerts:**

* 4 for Potentially overflowing call to snprintf
* 3 for Comparison result is always the same
* 2 for Multiplication result converted to larger type

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by keszybz** [comment]
I forgot to update the test. Updated.

**Comment by poettering** [comment]
lgtm

**Comment by evverx** [comment]
This pull request **fixes 9 alerts** when merging 91db8ed5b2e67abf738381a6ed6a05a8271498cd into 14f37112c86b573cf7f1f557e6d1be4c322290a4 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-c22f306b3a3d4433850e6d8868d19827e6c779ed)

**fixed alerts:**

* 4 for Potentially overflowing call to snprintf
* 3 for Comparison result is always the same
* 2 for Multiplication result converted to larger type

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by evverx** [comment]
@keszybz as far as I know, it's possible to sign in with your GitHub account at https://competitions.lgtm.com/ghu-2018 and some money will be donated to to the World Wildlife Fund for all the alerts you fixed.

File: src/journal/journal-verify.c [hunk 1]
```diff
@@ -60,10 +60,11 @@ static void draw_progress(uint64_t p, usec_t *last_usec) {
...
 
 static uint64_t scale_progress(uint64_t scale, uint64_t p, uint64_t m) {
+        /* Calculates scale * p / m, but handles m == 0 safely, and saturates.
+         * Currently all callers use m >= 1, but we keep the check to be defensive.
+         */
 
-        /* Calculates scale * p / m, but handles m == 0 safely, and saturates */
-
-        if (p >= m || m == 0)
+        if (p >= m || m == 0) /* lgtm [cpp/constant-comparison] */
```
Comment (evverx) [inline src/journal/journal-verify.c:67]: It seems that the alert this comment was supposed to suppress is still [there](https://lgtm.com/projects/g/systemd/systemd/alerts/?mode=tree&ruleFocus=2154840804). I'm not sure what's going on. Maybe it should start with `//` to be treated specially.

**Comment by keszybz** [comment]
> sign in with your GitHub account

Done.

> Maybe it should start with // to be treated specially.

It seems it has to be at the end of a line.
https://discuss.lgtm.com/t/rfe-support-for-supression-comments-in-comment-blocks/1364

**Pull request merged**